### PR TITLE
Make keyboard dark on dark themes (iOS only)

### PR DESCRIPTION
**Pull request by cardoso**
#### Summary
This will pass the `keyboardAppearance` prop into every TextInput that is themed. It's auto-detected from the current theme's `centerChannelBg` using the same approach used for the satus bar color. It makes the keyboard `'light'` or `'dark'` on iOS. React Native hasn't implemented this effect on Android yet as of 0.60. ([ref](https://facebook.github.io/react-native/docs/textinput#keyboardappearance)) I'm still passing the props into Android-only components/files in case they start supporting it (lmk if not including is a better practice for mattermost)

#### Checklist
- [x] Has UI changes

#### Device Information
This PR was tested on: [
    iPhone SE iOS 12.2,
    iPhone 5s iOS 12.2 Simulator,
    Google Nexus 5X Genymotion Simulator API Version 24
] 

#### Screenshots
| Light Theme (iOS) | Dark Theme (iOS) |
|-|-|
|Screen Shot 2019-08-10 at 10 14 22 AM|Screen Shot 2019-08-10 at 10 14 34 AM|

### CODE REVIEW

File: app/utils/theme.js [hunk 1]
```diff
@@ -42,3 +44,7 @@ export function isThemeSwitchingEnabled(state) {
     const {config} = state.entities.general;
     return config.EnableThemeSelection === 'true';
 }
+
+export function getKeyboardAppearanceFromTheme(theme) {
+    return tinyColor(theme.centerChannelBg).isLight() ? 'light' : 'dark';
```
Comment (enahum) [inline app/utils/theme.js]: Hey @cardoso please add a unit test for this function

Comment (cardoso) [reply app/utils/theme.js]: Nice catch! Done :)

**Comment by saturninoabril** [comment]
@cardoso Looks good.  One thing I observed, if I changed theme from other client (like browser), the UI gets updated except for the keyboard.  New theme is applied only when I closed then opened the keyboard - https://youtu.be/HkNJHTKPekY

**Comment by enahum** [comment]
Saturnino, that is controlled by the OS and I doubt it will ever be possible to work unless we “auto” close/open the keyboard when that happens which to me is even weirdest

**Comment by enahum** [comment]
@esethna can you please review

**Pull request merged**